iio/iio.c

```diff
@@ -0,0 +1,797 @@
+
+	if (mask & ~ch_mask)
+		return -ENOENT;
+
+	dev->ch_mask = ch_mask;
+
+	return SUCCESS;
+}
+
+/**
+ * close device
+ * @param *device name
+ * @return 0 on success, otherwise negative value
+ */
+static int32_t iio_close_dev(const char *device)
+{
+	return iio_supporter_dev(device) ? SUCCESS : FAILURE;
+}
+
+/**
+ * get device mask, this specifies the channels that are used
+ * @param *device name
+ * @param *device mask
+ * @return 0 on success, otherwise negative value
+ */
+static int32_t iio_get_mask(const char *device, uint32_t *mask)
+{
+	struct iio_interface *dev;
+	uint32_t ch_mask;
+
+	if (!iio_supporter_dev(device))
+		return -ENODEV;
+
+	dev = iio_get_interface(device, iio_interfaces);
+	ch_mask = 0xFFFFFFFF >> (32 - dev->num_channels);
+	*mask = ch_mask;
+
+	return SUCCESS;
+}
+
+/**
+ * transfer_dev_to_mem data from ADC into RAM
+ * @param *device name
+ * @param bytes_count
+ * @return bytes_count
+ */
+static ssize_t iio_transfer_dev_to_mem(const char *device, size_t bytes_count)
+{
+	struct iio_interface *iio_interface = iio_get_interface(device, iio_interfaces);
+
+	if (iio_interface->transfer_dev_to_mem)
+		return iio_interface->transfer_dev_to_mem(iio_interface->dev_instance,
+				bytes_count);
+
+	return -ENOENT;
+}
+
+/**
+ * read data from RAM to pbuf, use "capture()" first
+ * @param *device name
+ * @param *buff where data's are stored
+ * @param *offset to the remaining data
+ * @param bytes_count
+ * @return bytes_count
+ */
+static ssize_t iio_read_dev(const char *device, char *pbuf, size_t offset,
+			    size_t bytes_count)
+{
+	struct iio_interface *iio_interface = iio_get_interface(device, iio_interfaces);
+
+	if(iio_interface->read_data)
+		return iio_interface->read_data(iio_interface->dev_instance, pbuf, offset,
+						bytes_count);
+
+	return -ENOENT;
+}
+
+/**
+ * transfer_mem_to_dev write data to DAC
+ * @param *device name
+ * @param *buff
+ * @param bytes_count
+ * @return bytes_count
+ */
+static ssize_t iio_transfer_mem_to_dev(const char *device, size_t bytes_count)
+{
+	struct iio_interface *iio_interface = iio_get_interface(device, iio_interfaces);
+
+	if (iio_interface->transfer_mem_to_dev)
+		return iio_interface->transfer_mem_to_dev(iio_interface->dev_instance,
+				bytes_count);
+
+	return -ENOENT;
+}
+
+/**
+ * write data to RAM
+ * @param *device name
+ * @param *buff
+ * @param *offset in memory, used if some data have been already written
+ * @param bytes_count
+ * @return bytes_count
+ */
+static ssize_t iio_write_dev(const char *device, const char *buf,
+			     size_t offset,  size_t bytes_count)
+{
+	struct iio_interface *iio_interface = iio_get_interface(device, iio_interfaces);
+	if(iio_interface->write_data)
+		return iio_interface->write_data(iio_interface->dev_instance, (char*)buf,
+						 offset, bytes_count);
+
+	return -ENOENT;
+}
+
+/**
+ * @brief Get a merged xml containing all devices.
+ * @param devs - physical device instance.
```

Comment:
Missing parameter - can you check again all the comments and add proper description for all parameters?